drivers/nuopc/ocn_stream_ndep.F90

```diff
@@ -0,0 +1,268 @@
+      nfiles = 0
+      do nf = 1,nfiles_max
+         if (stream_ndep_data_filename(nf) == '') then
+            exit
+         else
+            nfiles = nfiles + 1
+         end if
+      end do
+      allocate(stream_filenames(nfiles), stat=errstat)
+      if (errstat /= 0) then
+         write(lp,*) 'Failed to allocate stream_filenames'
+         if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) then
+            call ESMF_Finalize(endflag=ESMF_END_ABORT)
+         end if
+      end if
+      do nf = 1,nfiles
+         stream_filenames(nf) = trim(stream_ndep_data_filename(nf))
+      end do
+
+      ! Write out info
+      if (mnproc == 1) then
+         write(lp,'(a)'   ) ' '
+         write(lp,'(a,i8)')  'stream ndep settings:'
+         do nf = 1,nfiles
+            write(tmpchar,'(i0)') nf
+            write(lp,'(a,a)' )  '  stream_ndep_data_filename('//trim(tmpchar)//') = ',trim(stream_filenames(nf))
+         end do
+         write(lp,'(a,a)' )  '  stream_ndep_mesh_filename = ',trim(stream_ndep_mesh_filename)
+         write(lp,'(a,a,a)') '  stream_ndep_varnames      = ',trim(stream_ndep_varname(1)),trim(stream_ndep_varname(2))
+         write(lp,'(a,i8)')  '  stream_ndep_year_first    = ',stream_ndep_year_first
+         write(lp,'(a,i8)')  '  stream_ndep_year_last     = ',stream_ndep_year_last
+         write(lp,'(a,i8)')  '  stream_ndep_year_align    = ',stream_ndep_year_align
+         write(lp,'(a)'   )  ' '
+      endif
+
+      ! Initialize the cdeps data type sdat_ndep
+      call shr_strdata_init_from_inline(sdat_ndep,                &
+           my_task             = mnproc,                         &
+           logunit             = lp,                             &
+           compname            = 'OCN',                          &
+           model_clock         = model_clock,                    &
+           model_mesh          = model_mesh,                     &
+           stream_meshfile     = trim(stream_ndep_mesh_filename), &
+           stream_filenames    = stream_filenames,               &
+           stream_yearFirst    = stream_ndep_year_first,          &
+           stream_yearLast     = stream_ndep_year_last,           &
+           stream_yearAlign    = stream_ndep_year_align,          &
+           stream_fldlistFile  = stream_ndep_varname,             &
+           stream_fldlistModel = stream_ndep_varname,             &
+           stream_lev_dimname  = 'null',                         &
+           stream_mapalgo      = 'bilinear',                     &
+           stream_offset       = 0,                              &
+           stream_taxmode      = 'cycle',                        &
+           stream_dtlimit      = 1.0e30_r8,                      &
+           stream_tintalgo     = 'linear',                       &
+           stream_name         = 'Sea surface temperature ',     &
```

Comment:
I syuspect this will be changed? - `stream_name = Nitrogen deposition`

Reply:
Right! Thanks. I'm just starting to test this today.